### PR TITLE
test(kernels): add snapshot wave 14 for CPU kernel modules

### PR DESCRIPTION
**Pull request by EffortlessSteven**
## Snapshot Wave 14

Adds **33 insta snapshot tests** (35 .snap files) covering 5 CPU kernel modules in `bitnet-kernels` that lacked snapshot coverage:

### Modules covered

| Module | Tests | Coverage |
|--------|-------|----------|
| **residual** | 6 | add/scaled/dropout results + error messages |
| **concat** | 7 | output shapes, split shapes, concat values + errors |
| **dequant** | 9 | I2_S block/row/ternary output, pack round-trip + errors |
| **attention_mask** | 5 | causal/padding/sliding-window mask layouts |
| **gating** | 8 | GatingType Debug, SwiGLU/GeGLU/ReGLU output + errors |

### Verification
```
cargo test -p bitnet-kernels --test snapshot_wave14 --no-default-features --features cpu
# test result: ok. 33 passed; 0 failed
```

### CODE REVIEW

**Comment by EffortlessSteven** [comment]
Closing: stale test wave  significant drift from main, would need full rewrite to rebase.
